### PR TITLE
feat(content-distribution): sync post meta

### PR DESCRIPTION
**Pull request by miguelpeixe**
Implements support for syncing post meta when distributing content. The new `Newspack_Network\Content_Distribution::get_reserved_post_meta_keys()` should determine which keys to ignore on distribution, to be used by both `Outgoing_Post` when generating the payload and `Incoming_Post` when clearing the post meta before an update.

This approach supports unique and not unique post meta.

### Testing

1. Make sure you have a network setup with at least one hub and one node
2. Setup content distribution by creating a new post and running in wp shell:

```php
$post_id = 0; // Change to your post ID.
$outgoing_post = new Newspack_Network\Content_Distribution\Outgoing_Post( $post_id );
$outgoing_post->set_config( [ 'http://node1.local' ] ); // Change to match your node URL.
Newspack_Network\Content_Distribution::distribute_post( $outgoing_post );
```

3. Still in wp shell, update the post meta and distribute again:

```php
$post = $outgoing_post->get_post();
update_post_meta( $post->ID, 'single', 'value' ); // Single string
update_post_meta( $post->ID, 'array', [ 1, 2, 3 ] ); // Single array
add_post_meta( $post->ID, 'multiple', 'a' ); // Not unique meta
add_post_meta( $post->ID, 'multiple', 'b' ); // Not unique meta
Newspack_Network\Content_Distribution::distribute_post( $outgoing_post );
```

4. Once the events are processed, confirm the meta is updated correctly in the node:

```php
get_post_meta( $post_id );
// = [
//      "single" => [ "value" ],
//      "array" => [ "a:3:{i:0;i:1;i:1;i:2;i:2;i:3;}" ],
//      "multiple" => [ "a", "b", "c" ]
// ]
```

### CODE REVIEW

File: includes/incoming-events/class-network-post-updated.php [hunk 1]
```diff
@@ -37,11 +37,11 @@ public function process_in_node() {
 	 */
 	protected function process_post_updated() {
 		$payload = (array) $this->get_data();
-		$error   = Linked_Post::get_payload_error( $payload );
+		$error   = Incoming_Post::get_payload_error( $payload );
 		if ( is_wp_error( $error ) ) {
 			return;
 		}
-		$linked_post = new Linked_Post( $payload );
+		$linked_post = new Incoming_Post( $payload );
```
Comment (miguelpeixe) [inline includes/incoming-events/class-network-post-updated.php:44]: I forgot to update this on #160

File: includes/class-content-distribution.php [hunk 2]
```diff
@@ -109,6 +110,42 @@ public static function get_distributed_post_types() {
...
+		/**
+		 * Filters the reserved post meta keys that should not be distributed.
+		 *
+		 * @param string[] $reserved_keys The reserved post meta keys.
+		 * @param WP_Post  $post          The post object.
+		 */
+		$reserved_keys = apply_filters( 'newspack_network_content_distribution_reserved_post_meta_keys', $reserved_keys );
+
+		// Always preserve content distribution post meta.
+		$reserved_keys = array_merge(
```
Comment (naxoc) [inline includes/class-content-distribution.php]: Nit: we could also just return the array_merge here. No need for the `$reserved_key` var.

File: includes/content-distribution/class-outgoing-post.php [hunk 3]
```diff
@@ -189,4 +190,53 @@ function( $term ) {
...
+			[]
+		);
+
+		/**
+		 * Filters the post meta data for distribution.
+		 *
+		 * @param array   $meta The post meta data.
+		 * @param WP_Post $post The post object.
+		 */
+		$meta = apply_filters( 'newspack_network_distributed_post_meta', $meta, $this->post );
```
Comment (naxoc) [inline includes/content-distribution/class-outgoing-post.php]: Same with an un-needed var here

**Review by naxoc — APPROVED** [review]
👍 Looks good and tests well. I added some nits – feel free to ignore them : )

File: includes/class-content-distribution.php (hunk 2 above)
Comment (miguelpeixe) [reply includes/class-content-distribution.php]: Good call! Updated in 53de4f37d70e2beb1414cfb574d1bf2f3bf9000c

File: includes/content-distribution/class-outgoing-post.php (hunk 3 above)
Comment (miguelpeixe) [reply includes/content-distribution/class-outgoing-post.php]: Same in 53de4f37d70e2beb1414cfb574d1bf2f3bf9000c

**Comment by miguelpeixe** [comment]
Thanks for the review!

**Pull request merged**